### PR TITLE
Introduce example for "magic words" test type [v2]

### PR DESCRIPTION
**Pull request by clebergnu**
This is a very simple, hello world like, implementation of a resolver.
If given the right (magic) words, it will return a resolution for the
"magic" test type.

It also includes the simplest possible runner for "magic" words.
It'll result in a successfull test result for URI "pass", and a test
failure for URI "fail".  For anything else, it'll result in an error.

Signed-off-by: Cleber Rosa <crosa@redhat.com>

---

Changes from v1 (#4315):
 * Added section on Contributor’s Guide

### CODE REVIEW

File: docs/source/guides/contributor/chapters/plugins.rst [hunk 1]
```diff
@@ -179,3 +181,140 @@ Which extends the list of files to be parsed by settings object. Note this
...
+:class:`avocado.core.nrunner.BaseRunner` is used to create the command
+line application, which uses the previously implemented runner class
+for ``magic`` test types.
+
+.. literalinclude:: ../../../../../examples/plugins/tests/magic/avocado_magic/runner.py
+
+Activating the new test type plugins
+------------------------------------
+
+The plugins need to registered so that Avocado knows about it.  See
```
Comment (willianrampazzo) [inline docs/source/guides/contributor/chapters/plugins.rst:227]: s/need to registered/need to be registered/

File: examples/plugins/tests/README.rst [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+different types:
+
+ * resolvers: they resolve references into proper test descriptions
+   that Avocado can run
+
+ * runners: these make use of the resolutions made by resolvers and
+   actually execute the tests, reporting the results back to Avocado
+
+These are all based on the "nrunner" architecture.  To see them in
+efect, after enabling them with ``python setup.py develop --user``-like
```
Comment (willianrampazzo) [inline examples/plugins/tests/README.rst:15]: s/efect/effect/

**Review by willianrampazzo — COMMENTED** [review]
The code looks good to me. Feel free to force-push the syntax fixes.

**Review by willianrampazzo — APPROVED** [review]

**Comment by clebergnu** [comment]
Merging with the fixes suggested by @willianrampazzo, thanks!

**Pull request merged**